pkg/cmd/kafka/create/create.go

```diff
@@ -700,35 +723,107 @@ func promptKafkaPayload(opts *options, constants *remote.DynamicServiceConstants
+	return payload, nil
+}
+
+func marketplaceQuotaPrompt(orgQuota *accountmgmtutil.OrgQuotas, answers *promptAnswers, f *factory.Factory, err error) (*promptAnswers, error) {
+	validMarketPlaces := FetchValidMarketplaces(orgQuota.MarketplaceQuotas, answers.CloudProvider)
+	if len(validMarketPlaces) == 1 {
+		answers.Marketplace = validMarketPlaces[0]
+	} else {
+		marketplacePrompt := &survey.Select{
+			Message: f.Localizer.MustLocalize("kafka.create.input.marketplace.message"),
+			Options: validMarketPlaces,
 		}
-		// casting to nullable string
-		var clusterIdSNS kafkamgmtclient.NullableString
-		clusterIdSNS.Set(opts.selectedCluster.ClusterId)
+		err = survey.AskOne(marketplacePrompt, &answers.Marketplace)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-		var billingModelEnterprise kafkamgmtclient.NullableString
-		var enterprise = "enterprise"
-		billingModelEnterprise.Set(&enterprise)
+	if len(validMarketPlaces) > 0 {
 
-		var nullString kafkamgmtclient.NullableString
-		payload = &kafkamgmtclient.KafkaRequestPayload{
-			Name:                  answers.Name,
-			BillingModel:          billingModelEnterprise,
-			BillingCloudAccountId: nullString,
-			Marketplace:           nullString,
-			ClusterId:             clusterIdSNS,
-			//	set plan here
+		validMarketplaceAcctIDs := FetchValidMarketplaceAccounts(orgQuota.MarketplaceQuotas, answers.Marketplace)
+
+		if len(validMarketplaceAcctIDs) == 1 {
+			answers.MarketplaceAcctID = validMarketplaceAcctIDs[0]
+		} else {
+			marketplaceAccountPrompt := &survey.Select{
+				Message: f.Localizer.MustLocalize("kafka.create.input.marketplaceAccountID.message"),
+				Options: validMarketplaceAcctIDs,
+			}
+			err = survey.AskOne(marketplaceAccountPrompt, &answers.MarketplaceAcctID)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	return payload, nil
+	return answers, nil
+}
+
+func cloudProviderPrompt(f *factory.Factory, answers *promptAnswers) (*promptAnswers, error) {
+	cloudProviderNames, err := GetEnabledCloudProviderNames(f)
+	if err != nil {
+		return nil, err
+	}
+
+	cloudProviderPrompt := &survey.Select{
+		Message: f.Localizer.MustLocalize("kafka.create.input.cloudProvider.message"),
+		Options: cloudProviderNames,
+	}
+
+	err = survey.AskOne(cloudProviderPrompt, &answers.CloudProvider)
+	if err != nil {
+		return nil, err
+	}
+	return answers, nil
+}
+
+// This may need to altered as the `name` are mutable on ocm side
+func getClusterNameMap(f *factory.Factory, clusterList *kafkamgmtclient.EnterpriseClusterList) (*map[string]v1.Cluster, error) {
+	//	for each cluster in the list, get the name from ocm and add it to the cluster list
+	str := listclusters.CreateSearchString(clusterList)
+	ocmClusterList, err := clustermgmt.GetClusterListByIds(f, "", "", str, len(clusterList.Items))
+	if err != nil {
+		return nil, err
+	}
+	clusterMap := make(map[string]v1.Cluster)
+	for _, cluster := range ocmClusterList.Slice() {
+		clusterMap[cluster.ID()] = *cluster
```

Comment:
Isn't this supposed to be name?

Reply:
nope, the name is only coming from the ocm side, ID is common between both and immutable